### DISCUSSION TITLE
BTT Smart Filament Sensor only works once then gets disabled by Firmware.

### DISCUSSION THREAD

**Discussion by BlueWings172** [Jyers/Marlin#1344] [General Discussion]
Hello

I have recently installed BTT smart filament sensor and using auto-compiled 1.3.5 firmware with the below configuration:

[Filament Runout Distance] 10
[Filament Motion Sensor] True
[Sensor Pin State] High
Sensor checkbox checked on the display.

That said, there are two issues:

1-After you are notified that the filament ran out and asked by Octoprint to change the filament, the sensor is ignored from that point on and if the filament runs out again during that print there will be no action or notification. In fact, what is surprising is that the checkbox for the filament sensor in the firmware becomes unchecked after the initial filament runout notification.
Therefore, the sensor is automatically disabled and will not trigger in the next prints until the box is manually checked again which will be undone by the very next filament runout trigger. (yes, I have clicked on Store Settings when I enabled the sensor).

2- The filament runout notification is displayed on the LCD screen but you can only resume by clicking the message on Octoprint. Would be great if you can resume the print after changing the filament from the printer without having to go to the pc and resume from there.

Is this normal or intended behavior or am I missing something ?

Thanks for your help.

**Comment by BlueWings172** [comment]
I have tried the low setting first but it never worked. Also it seems that others in this forum have reported that only high works which was my experience.

I am using the latest firmware as I have auto-compiled it on 21 August which is after version 1.3.5b was released. When I check the sensor option in the display, store settings, turn the printer off and back on, the setting remains checked as it should. However, the firmware unchecks it when  the sensor is triggered and the sensor becomes disabled.

**Reply by Cal-LF** [reply]
@BlueWings172 Not sure if that is a feature but mine stays unchecked until I check it and use the runout sensor. It is unchecked when I restart the printer. The sensor will not trigger again during the same print job. @Jyers may have to verify if that is the way it is set up to work.

**Reply by Cal-LF** [reply]
I think the store settings is the key.
